dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/consumer/TaskPriorityQueueConsumer.java

```diff
@@ -151,6 +173,10 @@ public List<TaskPriority> batchDispatch(int fetchTaskNum) throws TaskPriorityQue
         List<TaskPriority> failedDispatchTasks = Collections.synchronizedList(new ArrayList<>());
         CountDownLatch latch = new CountDownLatch(fetchTaskNum);
 
+        if (!taskDispatchFailedQueue.isEmpty()) {
+            consumerThreadPoolExecutor.submit(() -> dispatchFailedBackToTaskPriorityQueue(fetchTaskNum));
+        }
```

Comment:
It's not good idea to consume the failed queue in the normal process.

BTW, this will cause the thread busy, you didn't wait the current batch task finished.

Reply:
Understood, I changed the processing location of the failed queue.